comit/src/expiries.rs

```diff
@@ -0,0 +1,591 @@
+}
+
+/// Possible next action for Alice.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum AliceAction {
+    // Happy path actions for Alice.
+    Create,
+    FundAlpha,
+    WaitForFundTransactionFinality,
+    WaitForBetaFunded,
+    RedeemBeta,
+    WaitForRedeemBetaTransactionFinality,
+    NoFurtherAction, // Implies swap is done from Alice's perspective.
+    // Cancel path.
+    Abort,         // Implies HTLC not funded, abort but take no further action.
+    WaitForRefund, // Implies alpha ledger HTLC funded but expiry not yet elapsed.
+    Refund,        // Implies HTLC funded and expiry time elapsed.
+}
+
+impl AliceState {
+    /// Gets the next action required to transition to the next state.
+    fn next(&self) -> (AliceAction, AliceState) {
+        use self::{AliceAction::*, AliceState::*};
+
+        match self {
+            None => (Create, Created),
+            Created => (FundAlpha, FundAlphaTransactionSent),
+            FundAlphaTransactionSent => (WaitForFundTransactionFinality, AlphaFunded),
+            AlphaFunded => (WaitForBetaFunded, BetaFunded),
+            BetaFunded => (RedeemBeta, RedeemBetaTransactionSent),
+            RedeemBetaTransactionSent => (WaitForRedeemBetaTransactionFinality, Done),
+            Done => (NoFurtherAction, Done),
+        }
+    }
+
+    /// The minimum time we need to allow to transition to the next state.
+    fn transition_period(&self, c: &Config) -> Duration {
+        use self::AliceAction::*;
+
+        let (next_action, _next_state) = self.next();
+
+        match next_action {
+            Create => {
+                // Transition from None to Created
+                c.alice_to_act()
+            }
+            FundAlpha => {
+                // Transition from Created to FundAlphaTransactionSent
+                c.alice_to_act()
+            }
+            WaitForFundTransactionFinality => {
+                // Transition from FundAlphaTransactionSent to AlphaFunded
+                c.include_alpha_fund_transaction() + c.finality_alpha()
+            }
+            WaitForBetaFunded => {
+                // Transition from AlphaFunded to BetaFunded
+                c.bob_to_act() + c.include_beta_fund_transaction() + c.finality_beta()
+            }
+            RedeemBeta => {
+                // Transition from BetaFunded to RedeemBetaTransactionSent
+                c.alice_to_act()
+            }
+            WaitForRedeemBetaTransactionFinality => {
+                // Transition from RedeemBetaTransactionSent to Done
+                c.include_beta_redeem_transaction() + c.finality_beta()
+            }
+            // Transitioning to any of the cancel actions takes, be definition, zero time.
+            NoFurtherAction | Abort | WaitForRefund | Refund => Duration::zero(),
+        }
+    }
+
+    /// True if Alice has funded the alpha HTLC. We return true as soon as the
+    /// fund transaction has been sent since any time after attempting to refund
```

Comment:
```suggestion
    /// fund transaction has been sent since any time after attempting to refund
```
I think it'd be good to reflect that in the name of the function as we have several states for a transaction (sent, mined, finalised).

Reply:
Used `has_sent_fund_transaction`